mailet/standard/src/test/java/org/apache/james/transport/mailets/StripAttachmentTest.java

```diff
@@ -19,58 +19,157 @@
+import org.apache.mailet.MailetException;
 import org.apache.mailet.base.test.FakeMail;
 import org.apache.mailet.base.test.FakeMailetConfig;
-import org.junit.Assert;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TemporaryFolder;
+
+import com.google.common.base.Charsets;
 
 public class StripAttachmentTest {
 
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    private TemporaryFolder folder;
+    private String folderPath;
+
+    @Before
+    public void setUp() throws IOException {
+        folder = new TemporaryFolder();
+        folder.create();
+        folderPath = folder.getRoot().getPath() + "/";
+    }
+
+    @After
+    public void tearDown() throws IOException {
+        folder.delete();
+    }
+
+    @Test
+    public void getBooleanParameterShouldReturnFalseWhenNullAndDefaultFalse() throws Exception {
+        boolean actual = StripAttachment.getBooleanParameter(null, false);
+        assertThat(actual).isFalse();
+    }
+    
+    @Test
+    public void getBooleanParameterShouldReturnTrueWhenTrueAndDefaultFalse() throws Exception {
+        boolean actual = StripAttachment.getBooleanParameter("true", false);
+        assertThat(actual).isTrue();
+    }
+    
+    @Test
+    public void getBooleanParameterShouldReturnTrueWhenYesAndDefaultFalse() throws Exception {
+        boolean actual = StripAttachment.getBooleanParameter("yes", false);
+        assertThat(actual).isTrue();
+    }
+
+    @Test
+    public void getBooleanParameterShouldReturnFalseWhenOtherAndDefaultFalse() throws Exception {
+        boolean actual = StripAttachment.getBooleanParameter("other", false);
+        assertThat(actual).isFalse();
+    }
+
+    @Test
+    public void getBooleanParameterShouldReturnTrueWhenNullAndDefaultTrue() throws Exception {
+        boolean actual = StripAttachment.getBooleanParameter(null, true);
+        assertThat(actual).isTrue();
+    }
+
+    @Test
+    public void getBooleanParameterShouldReturnFalseWhenNoAndDefaultTrue() throws Exception {
+        boolean actual = StripAttachment.getBooleanParameter("no", true);
+        assertThat(actual).isFalse();
+    }
+
+    @Test
+    public void getBooleanParameterShouldReturnFalseWhenFalseAndDefaultTrue() throws Exception {
+        boolean actual = StripAttachment.getBooleanParameter("false", true);
+        assertThat(actual).isFalse();
+    }
+
+    @Test
+    public void getBooleanParameterShouldReturnTrueWhenOtherAndDefaultTrue() throws Exception {
+        boolean actual = StripAttachment.getBooleanParameter("other", true);
+        assertThat(actual).isTrue();
+    }
+
     @Test
-    public void testSimpleAttachment() throws MessagingException, IOException {
+    public void serviceShouldNotModifyMailWhenNotMultipart() throws MessagingException, IOException {
         Mailet mailet = initMailet();
```

Comment:
The configuration is already tested (initShouldThrowWhenPatternAndNotPatternAreNull for example) and fully covered by tests.
There's a huge complexity on testing every combination, we decided to do unitary tests on methods instead.